Failure on macro splices in Scala 3 when RedundantBraces enabled

```
version = 3.0.0-RC3
runner.dialect = scala3
rewrite.rules = [RedundantBraces]
```

## Steps

Given code like this:
```scala
inline def test(inline v: String, inline pred: String): String =
    ${ testCode('v, 'pred)        }

  private def testCode(v: Expr[String], pred: Expr[String])(using Quotes): Expr[String] =
    Expr("abc")
```

When I run scalafmt like this:

```bash
scalafmt
```

## Problem

Scalafmt fails with:
```scala
[info] Formatting 6 Scala sources...
Router:2092 MISSING CASE:
[1229]$                              MacroSplice
[1231]                               Space
[1245]startsWithCode                 Ident
Router:2092 MISSING CASE:
[1529]$                              MacroSplice
[1531]                               Space
[1543]endsWithCode                   Ident
Router:2092 MISSING CASE:
[1798]$                              MacroSplice
[1800]                               Space
[1808]testCode                       Ident
[error] /.../ValidateString.scala:30: error: Unable to format file due to bug in scalafmt
[error]     ${ startsWithCode('v, 'pred)  }
[error]     ^: /.../ValidateString.scala
[error] (Compile / scalafmt) /.../ValidateString.scala:30: error: Unable to format file due to bug in scalafmt
[error]     ${ startsWithCode('v, 'pred)  }
[error]     ^: /.../ValidateString.scala
```

## Expectation

scalafmt does not fail

## Workaround

It works correctly without `RedundantBraces`

## Notes

It looks like `RedundantBraces` incorrectly removes the braces for macro splice and that causes problems

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala
@@ -298,6 +298,8 @@ class RedundantBraces(ftoks: FormatTokens) extends FormatTokensRewrite.Rule {
 
       case Term.Block(List(`b`)) => true
 
+      case _: Term.QuotedMacroExpr | _: Term.SplicedMacroExpr => false
+
       case _ =>
         settings.generalExpressions && shouldRemoveSingleStatBlock(b)
     }

--- a/scalafmt-tests/src/test/resources/scala3/Macro.stat
+++ b/scalafmt-tests/src/test/resources/scala3/Macro.stat
@@ -7,6 +7,12 @@ ${ locationImpl() }
 '{   new Location() }
 >>>
 '{ new Location() }
+<<< remove redundant braces in quote braces
+rewrite.rules = [RedundantBraces]
+===
+'{   { new Location() } }
+>>>
+'{ new Location() }
 <<< quote expr
 ${
 val x   = 'expr
@@ -27,11 +33,23 @@ ${ assertImpl('expr) }
 throw new   AssertionError(s"failed assertion: ${${ showExpr(expr) }}")
 >>>
 throw new AssertionError(s"failed assertion: ${${ showExpr(expr) }}")
-<<< functions
+<<< functions in splices
 '{   (x: T) => ${ f('x) } }
 >>>
 '{ (x: T) => ${ f('x) } }
-<<< unctions spliced
+<<< functions with RedundantBraces
+rewrite.rules = [RedundantBraces]
+===
+'{   (x: T) => ${ f('x) } }
+>>>
+'{ (x: T) => ${ f('x) } }
+<<< remove redundant braces in functions in splices
+rewrite.rules = [RedundantBraces]
+===
+'{   (x: T) => ${ { f('x) } } }
+>>>
+'{ (x: T) => ${ f('x) } }
+<<< functions spliced
   (x: Expr[T]) => '{ $f($x) }
 >>>
 (x: Expr[T]) => '{ $f($x) }
@@ -115,3 +133,15 @@ body match {
       } if aVeryLongCondition(parameter) => subArgs.flatMap(flatSumArgs)
   case body => body
 }
+<<< redundant braces in splices and quotes
+rewrite.rules = [RedundantBraces]
+===
+def to[T, R](f: Expr[T] => Expr[R])(using
+    t: Type[T]
+)(using Type[R], Quotes): Expr[T => R] =
+  '{ { (x: t.Underlying) => ${ { f('x) } } } }
+>>>
+def to[T, R](f: Expr[T] => Expr[R])(using
+    t: Type[T]
+)(using Type[R], Quotes): Expr[T => R] =
+  '{ (x: t.Underlying) => ${ f('x) } }

